src/libraries/System.Private.CoreLib/src/System/Buffers/Binary/Reader.cs

```diff
@@ -124,5 +118,23 @@ public static ulong ReverseEndianness(ulong value)
             return ((ulong)ReverseEndianness((uint)value) << 32)
                 + ReverseEndianness((uint)(value >> 32));
         }
+
+        // Note: ReverseEndianness(float) and ReverseEndianness(double)
+        // are not intended to be public APIs as these would produce invalid
+        // floating point numbers.
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static float ReverseEndianness(float value)
```

Comment:
This should not exist, even as internal API. You do not want to be ever operating with floats/doubles that have reversed endianess for the reasons described in the linked issue.

Reply:
Fixed